test/functional/admin/statistics_announcements_controller_test.rb

```diff
@@ -144,6 +144,43 @@ class Admin::StatisticsAnnouncementsControllerTest < ActionController::TestCase
     assert_equal Time.zone.now, announcement.cancelled_at
   end
 
+  view_test "POST :publish_cancellation re-renders cancellation form if changes are not valid" do
+    announcement = create(:statistics_announcement)
+    post :publish_cancellation,
+         params: {
+           id: announcement.id,
+           statistics_announcement: { cancellation_reason: "" },
+         }
+
+    assert_response :success
+    assert_select "ul.govuk-error-summary__list a[data-track-action='statistics announcement-error'][data-track-label=\"Cancellation reason must be provided when cancelling an announcement\"]", text: "Cancellation reason must be provided when cancelling an announcement"
```

Comment:
the success criteria of this and the test below `"PUT :update_cancel_reason re-renders cancel_reason form if changes are not valid" ` are identical.

It might be worth adding another assertion

```
assert_template :cancel

```

Reply:
Done ✅